components/datetime/src/fieldsets.rs

```diff
@@ -1363,22 +1334,22 @@ impl_zone_marker!(
     /// ```
     O,
     description = "UTC offset, long length",
-    length_override = Long,
+    sample_length = long,
     sample = "GMT-05:00",
-    field = (fields::TimeZone::LocalizedOffset, fields::FieldLength::Four),
     zone_essentials = yes,
 );
-
-impl_zone_marker!(
-    Os,
-    description = "UTC offset, short length",
-    length_override = Short,
-    sample = "GMT-5",
-    field = (fields::TimeZone::LocalizedOffset, fields::FieldLength::One),
-    zone_essentials = yes,
-);
-
-// TODO: Add short/long UTC offset?
+impl_marker_length_constructors!(O,);
+impl O {
+    pub(crate) fn to_field(self) -> (fields::TimeZone, fields::FieldLength) {
```

Comment:
Issue: `to_field` can be done in the macro. Previously I wrote it as a switch statement between long and non-long, with the two results being arguments to the macro.

Reply:
I didn't see a way to do this. `Zs`, `Vs`, `Z`, `V`, and `O` have clear mappings between the `Length` and the `FieldLength`, but `L` and `X` don't have a `Length` and produce different field length. This code is very easy to understand, wrapping it in multiple macros would not be.